compiler/modules/nback.llvm/llvm.bal

```diff
@@ -169,6 +170,25 @@ public class Module {
         }
     }
 
+    // Corresponds to LLVMAddGlobal
+    public function addGlobal(Type ty, string name) {
+        if self.globalVariables.hasKey(name) {
+            panic error(createLine(["Variable", name, "already", "declared"]));
```

Comment:
Does the C API panic in this case?

Our existing code for addFunctionDefn/Decl doesn't deal with adding something with the same name. We should be consistent here.

Also we should not allow a global function and global variable with the same name.

Probably need a combined `map<FunctionDefn|FunctionDecl|GlobalVariable>`.

Reply:
C api don't panic instead it appends to the name (for example if we add two variables named `g1` one will be `g1` other `g1.1`). But then when you need to get the second variable user must use the appended name (`g1.1`) otherwise (`g1`) they will get the first variable. I thought this is bit confusing to the user because they use one name to create variables and another to access them

Reply:
Same is true for functions as well but we don't provide get methods for functions so the user must be using the function object returned at the time of creation

Reply:
How is your code handling name clashes between global variables and functions?

I agree it's weird to add a function with one name, and then require a different name to retrieve it. But we need to be consistent in our treatment of functions, global variables and local variables. Seems to me easiest way is to eliminate `getNamedGlobal`.